controller/src/main/java/io/pravega/controller/store/kvtable/AbstractKVTableMetadataStore.java

```diff
@@ -128,6 +128,13 @@ public KeyValueTable getKVTable(String scope, final String name, OperationContex
         return Futures.completeOn(checkScopeExists(scope, context, executor)
                 .thenCompose(exists -> {
                     if (exists) {
+                        checkScopeInDeletingTable(scope, context, executor)
+                                .thenCompose(exist -> {
+                                    if (exist) {
+                                        throw new IllegalArgumentException("Scope already in deleting state: " + scope);
+                                    }
```

Comment:
As discussed earlier, we should not be throwing an IllegalArgumentException here, but instead return a failed future with UnsupportedOperationException("create {} failed, as scope in deleting state")